src/promptflow-devkit/promptflow/_sdk/_orchestrator/run_submitter.py

```diff
@@ -85,10 +86,19 @@ def _run_bulk(self, run: Run, stream=False, **kwargs):
             run._resume_from = self._ensure_run_completed(run._resume_from)
         # Start trace
         if self._config.is_internal_features_enabled():
-            from promptflow.tracing._start_trace import start_trace
+            from promptflow.tracing._start_trace import is_collection_writeable, start_trace
+
+            logger.debug("start trace for flow run...")
```

Comment:
Can extract this to a function to be reused both in run submitter and test submitter?

Reply:
i think no, in test, we need `session` (used in chat-window) and `collection`; while in batch, we need `collection`, `run` (batch run id) and `attributes`, so extract looks strange for me.